Src/coffeescript/cql-execution/src/elm/quantity.coffee

```diff
@@ -19,24 +21,110 @@ module.exports.Quantity = class Quantity extends Expression
+      @value >= parseFloat other.value
+    else if other instanceof Quantity and ucum_time_units[other.unit]? and ucum_time_units[@unit]?
+      thisDurationInMilliseconds = durationInMilliseconds(@)
+      otherDurationInMilliseconds = durationInMilliseconds(other)
+      thisDurationInMilliseconds >= otherDurationInMilliseconds
+    else null
 
   after: (other) ->
-    if other instanceof Quantity and other.unit == @unit then @value > parseFloat other.value else null
+    if other instanceof Quantity and other.unit == @unit
+      @value > parseFloat other.value
+    else if other instanceof Quantity and ucum_time_units[other.unit]? and ucum_time_units[@unit]?
+      thisDurationInMilliseconds = durationInMilliseconds(@)
+      otherDurationInMilliseconds = durationInMilliseconds(other)
+      thisDurationInMilliseconds > otherDurationInMilliseconds
+    else null
 
   before: (other) ->
-    if other instanceof Quantity and other.unit == @unit then @value < parseFloat other.value else null
+    if other instanceof Quantity and other.unit == @unit
+      @value < parseFloat other.value
+    else if other instanceof Quantity and ucum_time_units[other.unit]? and ucum_time_units[@unit]?
+      thisDurationInMilliseconds = durationInMilliseconds(@)
+      otherDurationInMilliseconds = durationInMilliseconds(other)
+      thisDurationInMilliseconds < otherDurationInMilliseconds
+    else null
     
   equals: (other) ->
-    if other.unit? && other.value? then @unit == other.unit && @value == parseFloat other.value else null
+    if other instanceof Quantity and @unit == other.unit
+      @value == parseFloat other.value
+    else if other instanceof Quantity and ucum_time_units[other.unit]? and ucum_time_units[@unit]?
+      thisDurationInMilliseconds = durationInMilliseconds(@)
+      otherDurationInMilliseconds = durationInMilliseconds(other)
+      thisDurationInMilliseconds == otherDurationInMilliseconds
+    else null
 
-time_units = {'years':'year',  'months': 'month',  'days' :'day', 'minutes': 'minute', 'seconds':'seconds', 'milliseconds' : 'millisecond' }
 
 clean_unit = (units) ->
-  if time_units[units] then time_units[units] else units
+  if ucum_time_units[units] then ucum_to_cql_units[ucum_time_units[units]] else units
+
+# Hash of time units and their UCUM equivalents, both case-sensitive and case-insensitive
+# See http://unitsofmeasure.org/ucum.html#para-31
+# The CQL specification says that dates are based on the Gregorian calendar
+# UCUM says that years should be Julian. As a result, CQL-based year and month identifiers will
+# be matched to the UCUM gregorian units. UCUM-based year and month identifiers will be matched 
+# to the UCUM julian units.
+ucum_time_units = {'years': 'a_g', 'year': 'a_g', 'YEARS': 'a_g', 'YEAR': 'a_g', 'a_g': 'a_g'
+  , 'a': 'a_j', 'ANN': 'a_j', 'ann': 'a_j', 'A': 'a_j', 'a_j': 'a_j'
+  , 'months': 'mo_g', 'month':'mo_g', 'mo_g': 'mo_g'
+  , 'mo': 'mo_j', 'MO': 'mo_j', 'mo_j', 'mo_j'
```

Comment:
I think that `'MO': 'mo_j', 'mo_j', 'mo_j'` is intended to be: `'MO': 'mo_j', 'mo_j': 'mo_j'`, right?